Libs/MRML/Core/CMakeLists.txt

```diff
@@ -223,6 +223,8 @@ if(MRML_USE_vtkTeem)
     vtkMRMLNRRDStorageNode.cxx
     vtkMRMLTensorVolumeNode.cxx
     vtkMRMLVectorVolumeNode.cxx
+    vtkMRMLBitStreamVolumeNode.cxx
+    vtkMRMLCompressionDeviceNode.cxx
```

Comment:
Codec should not be a MRML node, as nodes are only intended to be used for data storage and not for processing. If we put the codec in a MRML node then it cannot have access to application or logic classes. Instead, we should define a simple vtkObject-based class as codec interface class, concrete codec classes (in the Slicer core we could have only one codec, a trivial "Uncompressed" codec that just copies data), and a codec factory where you can register custom codecs. Similarly how it is done in segmentation converter rules: http://apidocs.slicer.org/master/classvtkSegmentationConverterFactory.html

Volumes logic class would contain the codec factory. When a streaming volume node is added to the scene, volumes logic would read the codec name from the node, instantiate the corresponding codec class, and set it in the streaming volume node.

Reply:
Great Suggestion!
Should i create a folder "vtkStreamingVideoCore" in the Slicer/Libs, same level as the vtkSegmentationCore folder?
And place the vtkStreamingVideoCodec interface class, a trival "Uncompressed" codec class and the vtkStreamingVideoCodecFactory in the folder?

Reply:
Probably the interface class should be call vtkStreamingVolumeCodec. and the folder "vtkStreamingVolumeCore", as the data might be normal volume instead of Video

Reply:
Code in Slicer/Libs cannot use any MRML or Slicer module logic, so only the codec abstract class and uncompressed codec class could be there. Adding a new library to Slicer/Libs is quite a lot of small changes, so just for 2 classes I would not do it.

These codecs are only for image encoding, so I would put them in Libs/MRML/Core and/or Volumes module MRML and Logic folders.

Reply:
> vtkStreamingVolumeCodec

OK. Even if the most commonly used codecs are for 2D+t video data, the infrastructure would support streaming of volumes.

Reply:
Addressed